internal/message_queue/nats/client/nats_jetstream/nats_jetstream_client.go

```diff
@@ -0,0 +1,210 @@
+package nats_jetstream
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+	"google.golang.org/protobuf/proto"
+)
+
+type Client struct {
+	js          jetstream.JetStream
+	nc          *nats.Conn
+	logger      *slog.Logger
+	consumers   map[string]jetstream.Consumer
+	storageType jetstream.StorageType
+	ctx         context.Context
+	cancelAll   context.CancelFunc
+}
+
+var (
+	ErrConsumerNotInitialized = errors.New("consumer for topic not initialized")
+)
+
+func WithSubscribedTopics(topics ...string) func(handler *Client) error {
+	return func(c *Client) error {
+		for _, topic := range topics {
+			streamMinedTxs, err := c.getStream(topic, fmt.Sprintf("%s-stream", topic))
+			if err != nil {
+				return fmt.Errorf("failed to get stream: %v", err)
+			}
+
+			cons, err := c.getConsumer(streamMinedTxs, fmt.Sprintf("%s-cons", topic))
+			if err != nil {
+				return fmt.Errorf("failed to get consumer: %v", err)
+			}
+
+			c.consumers[topic] = cons
+		}
+
+		return nil
+	}
+}
+
+func WithFileStorage() func(handler *Client) error {
+	return func(c *Client) error {
+		c.storageType = jetstream.FileStorage
+		return nil
+	}
+}
+
+type Option func(p *Client) error
+
+func New(nc *nats.Conn, logger *slog.Logger, topics []string, opts ...Option) (*Client, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	p := &Client{
+		logger:      logger,
+		nc:          nc,
+		consumers:   map[string]jetstream.Consumer{},
+		storageType: jetstream.MemoryStorage,
+		ctx:         ctx,
+		cancelAll:   cancel,
+	}
+
+	js, err := jetstream.New(nc)
+	if err != nil {
+		return nil, err
+	}
+
+	p.js = js
+
+	for _, topic := range topics {
+		_, err = p.getStream(topic, fmt.Sprintf("%s-stream", topic))
```

Comment:
What's the point of this call? Just checking if we could encounter error?

Reply:
The `getStream()` function returns the stream, but also ensures it exists i.e. it returns the existing one or creates it if it not exists. For a pure publisher the stream is not needed. It is only needed for a subscriber. Therefore there is no stream returned.